src/pages/ConvertVrt/Convert/index.tsx

```diff
@@ -1,141 +1,218 @@
-        <AmountForm onSubmit={noop} maxAmount={vrtLimitWei.toFixed()} css={styles.form}>
-          {({ values }) => {
-            const xvsValue = values.amount
-              ? new BigNumber(values.amount).times(xvsToVrtRate).dp(VRT_DECIMAL).toFixed()
-              : '';
-            return (
-              <>
-                <div css={styles.inputSection}>
-                  <Typography variant="small2" css={styles.inputLabel}>
-                    {t('convertVrt.convertVrt')}
-                  </Typography>
-                  <FormikTokenTextField
-                    tokenId={VRT_ID}
-                    name="amount"
-                    css={styles.input}
-                    description={
-                      <Trans
-                        i18nKey="convertVrt.balance"
-                        components={{
-                          White: <span css={styles.whiteLabel} />,
-                        }}
-                        values={{ amount: format(new BigNumber(vrtLimitWei)) }}
-                      />
+        <EnableToken
+          title={t('convertVrt.enableVrt')}
+          assetId={VRT_ID}
+          isEnabled={userVrtEnabled}
+          tokenInfo={[
+            {
+              iconName: 'xvs',
+              label: 'VRT Conversion Ratio',
+              children: xvsToVrtConversionRatio?.toFixed(6),
+            },
+            { iconName: 'vrt', label: 'Current VRT Balance', children: readableUserVrtBalance },
+          ]}
+          vtokenAddress={vrtConverterProxyAddress}
+        >
+          <AmountForm onSubmit={onSubmit} maxAmount={userVrtBalance} css={styles.form}>
+            {({ values }) => {
+              const xvsValue =
+                values.amount && xvsToVrtConversionRatio
+                  ? new BigNumber(values.amount)
+                      .times(xvsToVrtConversionRatio)
+                      .dp(VRT_DECIMAL)
+                      .toFixed()
+                  : '';
+              return (
+                <>
+                  <div css={styles.inputSection}>
+                    <Typography variant="small2" css={[styles.inputLabel, styles.fontWeight600]}>
+                      {t('convertVrt.convertVrt')}
+                    </Typography>
+                    <FormikTokenTextField
+                      tokenId={VRT_ID}
+                      name="amount"
+                      css={styles.input}
+                      description={
+                        <Trans
+                          i18nKey="convertVrt.balance"
+                          components={{
+                            White: <span css={styles.whiteLabel} />,
+                          }}
+                          values={{ amount: readableUserVrtBalance }}
+                        />
+                      }
+                      rightMaxButton={{
+                        label: t('convertVrt.max').toUpperCase(),
+                        valueOnClick: userVrtBalance || '',
```

Comment:
If a user clicks on the max button and it doesn't nothing, they'd think it's broken. I think we should just hide the max button if `userVrtBalance` is undefined